src/utilcode/util.cpp

```diff
@@ -637,8 +637,15 @@ BYTE * ClrVirtualAllocWithinRange(const BYTE *pMinAddr,
             if (!ClrVirtualQuery((LPCVOID)tryAddr, &mbInfo, sizeof(mbInfo)))
                 break;
             
+#if defined(FEATURE_PAL) && !defined(HAVE_VM_ALLOCATE)
```

Comment:
I believe that PAL returns region size as 0 in this case. It may be better to just check for it instead of fragile ifdef: 

((mbInfo.RegionSize >= (SIZE_T) dwSize) || (mbInfo.RegionSize == 0))